docs/concepts/controller-manager.md

```diff
@@ -278,6 +278,32 @@ Since the other two reconcilers are unable to actively monitor the relevant obje
 
 The `Project Activity Reconciler` is implemented to take care of such cases. An event handler will notify the reconciler for any activity and then it will update the `status.lastActivityTimestamp`. This update will also trigger the `Stale Project Reconciler`.
 
+
+#### [`ResourceQuota` Reconciler](../../pkg/controllermanager/controller/project/resourcequota)
+
+The `ResourceQuota` reconciler only reconciles `ResourceQuota`s in `Project` namespaces and ensures that the specified quotas do not interfere with Gardener's operations.
+The reconciler gets also triggered when a `Shoot` in the respective `Project` namespace is created.
+
+In the lifecycle of a `Shoot`, Gardener also creates some resources (`Secret`s and `ConfigMap`s) in the `Project` namespace. This means that the `ResourceQuota` must allow for the creation of these resources, otherwise the `Shoot` reconciliation would fail due to quota violations.
+
+The reconciler checks `ResourceQuota`s in the `Project` namespace and increases the limits according to the following rules:
+
+##### `count/shoots.core.gardener.cloud` limit present
+
+When the `ResourceQuota` contains a limit for `count/shoots.core.gardener.cloud` ("Shoot limit"), the reconciler increases the following limits based on that value:
+
+- `count/secrets`: set to at least `Shoot limit * maximum Secrets per Shoot`
+- `count/configmaps`: set to at least `Shoot limit * maximum ConfigMaps per Shoot`
+
+##### `count/shoots.core.gardener.cloud` limit absent
+
+When there is not `Shoot` limit specified, the controller sets the following limits to fit all current `Shoot`s:
+
+- `count/secrets`: set to at least `current number of Shoots * maximum Secrets per Shoot`
+- `count/configmaps`: set to at least `current number of Shoots * maximum ConfigMaps per Shoot`
+
+When the `ResourceQuota` does not contain a limit for `count/shoots.core.gardener.cloud`, the reconciler increases the following limits to a default minimum value.
```

Comment:
Is this sentence a leftover? It seems disconnected or redundant to what was already clarified above.